plugins/woocommerce-admin/client/products/fills/more-menu-items/delete-variation-menu-item.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * External dependencies
+ */
+import { useParams } from 'react-router-dom';
+import { MenuGroup, MenuItem } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { useMemo, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME,
+	ProductVariation,
+	ProductVariationAttribute,
+} from '@woocommerce/data';
+import { getNewPath, navigateTo } from '@woocommerce/navigation';
+import { RemoveConfirmationModal } from '@woocommerce/product-editor';
+import { recordEvent } from '@woocommerce/tracks';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { useEntityId, useEntityProp } from '@wordpress/core-data';
+
+export type DeleteVariationMenuItemProps = { onClose(): void };
+
+export const DeleteVariationMenuItem = ( {
+	onClose,
+}: DeleteVariationMenuItemProps ) => {
+	const [ showModal, setShowModal ] = useState( false );
+
+	const { productId } = useParams();
+
+	const variationId = useEntityId( 'postType', 'product_variation' );
+
+	const [ attributes ] = useEntityProp< ProductVariationAttribute[] >(
+		'postType',
+		'product_variation',
+		'attributes'
+	);
+
+	const [ status ] = useEntityProp< string >(
+		'postType',
+		'product_variation',
+		'status'
+	);
+
+	const name = useMemo(
+		() =>
+			attributes
+				.map(
+					( attribute: ProductVariationAttribute ) => attribute.option
+				)
+				.join( ', ' ),
+		[ attributes ]
+	);
```

Comment:
@octaedro in his title PR exposed the `name` attribute in the variation API, so you should just be able to do:
```
const [ attributes ] = useEntityProp< ProductVariationAttribute[] >(
		'postType',
		'product_variation',
		'name'
	);
```
Instead of creating it yourself out off the attributes. This will also help a lot if we want to make this component more re-usable across multiple product types.

Reply:
right! thanks for recalling this